### PR TITLE
feat: stylize text inside backticks when appearing in example description

### LINKED ISSUE #253: Stylize text inside `backticks` when appearing in example description
The python client does not stylize text inside tilde when it appears in the example description (please see screenshot). The rust client **tlrc** does stylize text when it appears inside ``

The python client needs to stylize text inside `` when it appears in the description.

Screenshot from 2024-09-06 20-45-25

### PR DESCRIPTION
**Pull request by kyluca**
Closes #253 

Hoping I've understood the brief correctly :)

This change will check for the presence of backticks (`) and replace them with the ANSI escape characters for italics (and a matching reset to close the pairs).

This appears to match the desired output from the Rust CLI from the issue.

Example output:
screenshot_from_2024_10_02_16_50_05

### CODE REVIEW

**Review by owenvoke — COMMENTED** [review]
Hi @kyluca, thanks for looking into this. It looks like this seems to reset the styling after the back ticks (it resets from being green). Would we be able to reapply the styles after this? 🤔

**Comment by kyluca** [comment]
Ah of course, good catch 👍

**Comment by kyluca** [comment]
Updated to reapply default or user styles, as well as a hardcoded yellow for further emphasis (matching Rust client output).

Default style:
screenshot_from_2024_10_02_23_40_43

Custom style: To show I didn't cheat and use "green" as the reset haha
screenshot_from_2024_10_02_23_41_03

File: tldr.py [hunk 1]
```diff
@@ -414,6 +417,11 @@ def colors_of(key: str) -> Tuple[str, str, List[str]]:
 
 
 def output(page: str, plain: bool = False) -> None:
+    def emphasise_example(x: str) -> str:
```
Comment (kyluca) [inline tldr.py:420]: Opted for a `def` to avoid a `lambda` exceeding the line length.

File: tldr.py [hunk 2]
```diff
@@ -422,23 +430,48 @@ def output(page: str, plain: bool = False) -> None:
...
+        # Handle an example description
         elif line[0] == '-':
-            line = '\n' + ' ' * LEADING_SPACES_NUM + \
-                colored(line, *colors_of('example'))
+
+            # Stylize text within backticks using yellow italics
+            if '`' in line:
+                elements = ['\n', ' ' * LEADING_SPACES_NUM]
+
+                for item in EXAMPLE_SPLIT_REGEX.split(line):
```
Comment (kyluca) [inline tldr.py:459]: Followed the pattern for example commands below, using regex to pull out the substrings using backticks.

**Review by owenvoke — APPROVED** [review]
Looks great to me, but I'll leave this for someone with more Python knowledge to merge. 👍🏻

**Comment by kyluca** [comment]
I noted the builds are failing due to an unrelated error, so have put up a separate PR to fix them first: https://github.com/tldr-pages/tldr-python-client/pull/256

**Pull request merged**